src/coreComponents/physicsSolvers/wavePropagation/AcousticVTIWaveEquationSEM.cpp

```diff
@@ -0,0 +1,1027 @@
+                                        arrayView1d< string const > const & GEOS_UNUSED_PARAM ( regionNames ) )
+  {
+    NodeManager & nodeManager = mesh.getNodeManager();
+
+    arrayView1d< real32 > const p_nm1 = nodeManager.getField< fields::Pressure_p_nm1 >();
+    arrayView1d< real32 > const p_n = nodeManager.getField< fields::Pressure_p_n >();
+    arrayView1d< real32 > const p_np1 = nodeManager.getField< fields::Pressure_p_np1 >();
+
+    arrayView1d< real32 > const q_nm1 = nodeManager.getField< fields::Pressure_q_nm1 >();
+    arrayView1d< real32 > const q_n = nodeManager.getField< fields::Pressure_q_n >();
+    arrayView1d< real32 > const q_np1 = nodeManager.getField< fields::Pressure_q_np1 >();
+
+    if( computeGradient )
+    {
+
+      arrayView1d< real32 > const p_dt2 = nodeManager.getField< fields::PressureDoubleDerivative >();
+
+      if( NULL == std::getenv( "DISABLE_LIFO" ) )
+      {
+        m_lifo->pushWait();
+      }
+      forAll< EXEC_POLICY >( nodeManager.size(), [=] GEOS_HOST_DEVICE ( localIndex const nodeIdx )
+      {
+        p_dt2[nodeIdx] = (p_np1[nodeIdx] - 2*p_n[nodeIdx] + p_nm1[nodeIdx])/(dt*dt);
+      } );
+
+      if( NULL == std::getenv( "DISABLE_LIFO" ) )
+      {
+        // Need to tell LvArray data is on GPU to avoir HtoD copy
+        p_dt2.move( MemorySpace::cuda, false );
+        m_lifo->pushAsync( p_dt2 );
+      }
+      else
+      {
+        GEOS_MARK_SCOPE ( DirectWrite );
+        p_dt2.move( MemorySpace::host, false );
+        int const rank = MpiWrapper::commRank( MPI_COMM_GEOSX );
+        std::string fileName = GEOS_FMT( "lifo/rank_{:05}/pressuredt2_{:06}_{:08}.dat", rank, m_shotIndex, cycleNumber );
+        int lastDirSeparator = fileName.find_last_of( "/\\" );
+        std::string dirName = fileName.substr( 0, lastDirSeparator );
+        if( string::npos != (size_t)lastDirSeparator && !dirExists( dirName ))
+          makeDirsForPath( dirName );
+
+        //std::string fileName = GEOS_FMT( "pressuredt2_{:06}_{:08}_{:04}.dat", m_shotIndex, cycleNumber, rank );
+        //const int fileDesc = open( fileName.c_str(), O_CREAT | O_WRONLY | O_DIRECT | O_TRUNC, S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP |
+        // S_IROTH | S_IWOTH );
+
+        std::ofstream wf( fileName, std::ios::out | std::ios::binary );
+        GEOS_THROW_IF( !wf,
+                        "Could not open file "<< fileName << " for writting",
+                        InputError );
+        wf.write( (char *)&p_dt2[0], p_dt2.size()*sizeof( real32 ) );
+        wf.close( );
+        GEOS_THROW_IF( !wf.good(),
+                        "An error occured while writting "<< fileName,
+                        InputError );
+      }
+
+    }
+
+    forAll< EXEC_POLICY >( nodeManager.size(), [=] GEOS_HOST_DEVICE ( localIndex const a )
+    {
+      p_nm1[a] = p_n[a];
+      p_n[a]   = p_np1[a];
+
+      q_nm1[a] = q_n[a];
+      q_n[a]   = q_np1[a];
+
+    } );
+  } );
+
+  return dtOut;
+}
+
+
```

Comment:
We can keep it as it is but... not sure that for now the backward step is necessary + not sure that the formula inside it are correct for VTI anyway... Maybe we should remove it for now what do you think @sframba ?

Reply:
I let the Backward function as I need to (pure virtual) but the function is now just sending an error. Is that ok?

Reply:
You can do the same thing as in other solvers, in the Elastic one we do as follow : 
`real64 ElasticWaveEquationSEM::explicitStepBackward( real64 const & time_n,
                                                     real64 const & dt,
                                                     integer cycleNumber,
                                                     DomainPartition & domain,
                                                     bool GEOS_UNUSED_PARAM( computeGradient ) )
{
  GEOS_ERROR( "Backward propagation for the elastic wave propagator not yet implemented" );
  real64 dtOut = explicitStepInternal( time_n, dt, cycleNumber, domain );
  return dtOut;
}`

And let the declaration in the hpp intact

Reply:
Yes, that's what I did

`GEOS_ERROR( "This option is not supported yet" );`